drivers/wifi/esp32/src/esp_wifi_drv.c

```diff
@@ -378,22 +465,26 @@ static int esp32_wifi_connect(const struct device *dev,
 		return -EAGAIN;
 	}
 
-	if (mode != ESP32_WIFI_MODE_STA) {
+	if (IS_ENABLED(CONFIG_ESP32_WIFI_AP_STA_MODE) &&
+	    (mode == ESP32_WIFI_MODE_AP || mode == ESP32_WIFI_MODE_APSTA)) {
+		ret = esp_wifi_set_mode(ESP32_WIFI_MODE_APSTA);
+	} else {
 		ret = esp_wifi_set_mode(ESP32_WIFI_MODE_STA);
```

Comment:
Should it handles the case where only AP mode is enabled? I mean, `mode` is not checked if L468 condition is false.

Reply:
While implementing this condition, my logic was,

If AP_STA mode is disabled then there will be only one iface  to manage and if someone tries to connect to Wi-Fi and no matter what the current mode is, put the controller in STA mode, So it can try to connect to Wi-Fi. No need to manage the different modes in this case, to keep things simple.

Looking for your thoughts.